src/E-Shop.Api/Startup.cs

```diff
@@ -43,6 +44,7 @@ public void ConfigureServices(IServiceCollection services)
                  optionsBuilder.MigrationsAssembly("E-Shop.Api")));
             services.AddScoped<ICategoryRepository,CategoryRepository>();
             services.AddScoped<ICategoryServices,CategoryServices>();
+            services.AddScoped(typeof(IBaseRepository<>),typeof(BaseRepository<>));
```

Comment:
why using `typeof`? we can use `services.AddScoped(IBaseRepository,BaseRepository);`

Reply:
I'm getting error when i set services.AddScoped(IBaseRepository,BaseRepository);.
using typeof solves my issue.Thats why i used typeof

Reply:
ok